src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -266,7 +266,13 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
         $fragment = isset($extra['_fragment']) ? $extra['_fragment'] : '';
         unset($extra['_fragment']);
 
-        if ($extra && $query = http_build_query($extra, '', '&')) {
+        if (defined('PHP_QUERY_RFC3986')) {
```

Comment:
The reason I asked was that you submitted the PR on master :) If this is a bug fix, this should be merged on 2.7 instead first.